src/main/java/duke/command/CommandKey.java

```diff
@@ -0,0 +1,36 @@
+package duke.command;
+
+/**
+ * Represents the list of supported command keywords.
+ */
+public enum CommandKey {
+    DONE("done"),
+    LIST("list"),
+    DELETE("delete"),
+    TODO("todo"),
+    EVENT("event"),
+    DEADLINE("deadline"),
+    EXIT("bye"),
+    FIND("find");
+    
+    private String key;
+    
+    CommandKey(String key) {
+        this.key = key;
+    }
+    
+    String getKey() {
+        return key;
+    }
+
+    /**
+     * Checks if the user's command is valid.
+     * 
+     * @param userCommand User's command keyword (first word in the user's input).
+     * @param key Supported command key.
+     * @return Whether user's command keyword is supported.
+     */
```

Comment:
Should this method just be called "equals"?
This method already belongs to the CommandKey enum, so it would be called as CommandKey.equalsCommandKey, which might be redundant.
Just having "equals" would also match the equals method for other data types like Strings